src/test/java/persistence/study/ReflectionTest.java

```diff
@@ -0,0 +1,135 @@
+import static persistence.study.Car.TEST_METHOD_PREFIX;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class ReflectionTest {
+
+	private static final Logger logger = LoggerFactory.getLogger(ReflectionTest.class);
+	private Class<Car> carClass;
+
+	@BeforeEach
+	void setUp() {
+		carClass = Car.class;
+	}
+
+	@Test
+	@DisplayName("Car 객체 정보 가져오기")
+	void showClass() {
+		logger.debug(carClass.getName());
+	}
+
+	@Nested
+	@DisplayName("메서드 테스트")
+	class MethodTest {
+		private Car car;
+		private Method[] declaredMethods;
+
+		@BeforeEach
+		void setUp() throws Exception {
+			// given
+			car = createCarInstance();
+			declaredMethods = carClass.getDeclaredMethods();
+		}
+
+		@Test
+		@DisplayName("Car 메서드 네임이 test로 시작하는 메서드 실행하기")
+		void executeMethodStartsWithTest() throws Exception {
+			// given
+			final String targetMethodName = "test";
+
+			// when
+			for (Method declaredMethod : declaredMethods) {
+				if (declaredMethod.getName().startsWith(targetMethodName)) {
+					final String result = (String) declaredMethod.invoke(car);
+					assertThat(result).startsWith(TEST_METHOD_PREFIX);
+				}
+			}
+		}
+
+		@Test
+		@DisplayName("@PrintView 애노테이션 메소드 실행하기")
+		void executeMethodWithPrintView() throws Exception {
+			// when
+			for (Method declaredMethod : declaredMethods) {
+				if (declaredMethod.isAnnotationPresent(PrintView.class)) {
+					declaredMethod.invoke(car);
+				}
+			}
+		}
+	}
+
+	@Nested
+	@DisplayName("필드 테스트")
+	class FieldTest {
+
+		private final String carName = "포르쉐";
+		private final int price = 1000;
+
+		@Test
+		@DisplayName("private field에 값 할당")
+		void privateFieldAccess() throws Exception {
+			final Car car = createCarInstance();
+
+			// when
+			this.injectField(car, "name", carName);
+			this.injectField(car, "price", price);
+
+			// then
+			this.assertResult(car);
+		}
+
+
+		@Test
+		@DisplayName("요구사항 5 - 인자를 가진 생성자의 인스턴스 생성")
+		void constructorWithArgs() throws Exception {
+			// given
+			final Constructor<?>[] constructors = carClass.getConstructors();
+			final int targetConstructorParameterNumber = 2;
+			Car car = null;
+
+			// when
+			for (Constructor<?> constructor : constructors) {
+				if (constructor.getParameterCount() == targetConstructorParameterNumber) {
+					car = (Car) constructor.newInstance(carName, price);
+					break;
+				}
+			}
+			if (car == null) {
+				throw new IllegalArgumentException("car 인스턴스 생성에 실패했습니다.");
+			}
+
+			// then
+			this.assertResult(car);
+		}
+
+
+		private void assertResult(final Car car) {
+			assertThat(car.testGetPrice()).isEqualTo(TEST_METHOD_PREFIX + price);
+			assertThat(car.testGetName()).isEqualTo(TEST_METHOD_PREFIX + carName);
+		}
```

Comment:
```suggestion
		private void assertResult(final Car car) {
			assertThat(car.testGetPrice()).isEqualTo("test: 1000");
			assertThat(car.testGetName()).isEqualTo("test: 포르쉐");
		}
```

그런 의미에서 `TEST_METHOD_PREFIX` 가 전역변수로 쓰여야 하는 이 테스트 방식은
제가 동료 개발자라고 가정했을 때 유지보수가 어려울 것 같습니다.

위의 assertResult 메서드는 파라미터로 주입받은 car 객체 외에,
`TEST_METHOD_PREFIX`, `price`, `carName` 에 대해서 의존성을 가지고 있습니다.
위의 테스트 의도를 파악하려면 car 객체가 이 3가지 외부 변수와 어떤 의존성이 있는지 파악하기 위해,
변수 선언을 확인하는 것을 넘어, 이 변수들이 어떻게 쓰여서 car 에 영향을 끼치는 지까지 로직을 파악해야합니다.

창천항로 이동욱 님은 이러한 현상을 막기 위해 ["검증부 (assert / expect)는 하드코딩한다"](https://jojoldu.tistory.com/615) 라고 블로그 글을 쓰신적이 있습니다.
저도 검증부는 하드코딩하는 스타일을 더 좋아하기 때문에,
미션을 진행하는 동안에는 assertThat 의 expected 를 가능한 하드코딩된 값을 넣어주시면 감사하겠습니다.

Reply:
검증부는 하드코딩한다 는 얘기는 이번에 처음 알게 되었네요! 글을 읽어보니 확실히 공감이 갑니다. 피드백 감사합니다.
이번 기회에 적용하도록 해보겠습니다.